pkg/apis/core/validation/validation.go

```diff
@@ -2732,6 +2732,40 @@ func ValidateVolumeDevices(devices []core.VolumeDevice, volmounts map[string]str
 	return allErrs
 }
 
+func validatePodResourceClaims(claims []core.PodResourceClaim, fldPath *field.Path) (sets.String, field.ErrorList) {
+	var allErrs field.ErrorList
+	podClaimNames := sets.NewString()
+	for i, claim := range claims {
+		allErrs = append(allErrs, validatePodResourceClaim(claim, &podClaimNames, fldPath.Index(i))...)
+	}
+	return podClaimNames, allErrs
+}
+
+func validatePodResourceClaim(claim core.PodResourceClaim, podClaimNames *sets.String, fldPath *field.Path) field.ErrorList {
```

Comment:
Not a blocker for code freeze: it's better to have basic prefix check for the ResourceClaims here use: 
https://github.com/kubernetes/kubernetes/blob/81bd2496bce080e5788cf18a807cdd42bd4ee8c1/staging/src/k8s.io/apimachinery/pkg/api/validation/path/name.go#L61-L68

It is just to check something like an invalid prefix like `.`/`..`/`/` exists .

Reply:
The code had full validation of the ResourceClaimName, but we took it out because it implied that core validation had to contain the validation function for ResourceClaim.Name - not the place where it belongs.

Using some lighter validation looks like a good compromise.

Reply:
My comment was about some other field, please ignore it.

For names inside spec.resourceClaims it makes sense to require that the names are simple - added, together with a test case.

Reply:
Instead of ValidatePathSegmentName, the even stricter ValidateDNS1123Label is now used in validatePodResourceClaim for the name.